### ISSUE TITLE
Cannot switch off NVIDIA GPU

### ISSUE THREAD

**Issue by balbok0** [dglt1/optimus-switch-sddm#12]
Hi,

I'm currently running Manjaro KDE Plasma, and switching between prime and intel modes works fine. My main problem is that NVIDIA GPU is not switched off properly when using `sudo set-intel.sh`. I tried following the guide in README but settings returning "working!" did not seem to switch off the GPU

<details>
<summary> Contents of cat /etc/switch/intel/no-optimus.sh </summary>

```
#!/bin/sh

#this script is run as a display setup script
#that replaces the one used by nvidia/prime.
##
#this powers-down/disables the nvidia dGPU and
#removes it from /sys/bus/pce/devices
#for the current boot. this is reset after reboot.
# PLEASE READ BELOW TO ENABLE THIS

xrandr --auto

#This ensures that LightDM doesn't fail after locking the screen or logging out
#edit "0000:01:00.0" to match your bus id if it's different.

if [ -d "/sys/bus/pci/devices/0000:01:00.0" ]
then

  echo 'auto' > '/sys/bus/pci/devices/0000:01:00.0/power/control'  #adjust busid if needed

fi



#to find out what acpi_call disables your nvidia gpu,
#run this command
#
#` sudo /usr/share/acpi_call/examples/turn_off_gpu.sh ` 
#
#and see which acpi_call is returned as "works!" and then edit line 38 to match if needed.
#default BusID is set to 0000:01:00.0 (syntax is important) 
#be sure to edit lines 16,19,36,38,39 to match your BusID and working acpi_call if needed
#then uncomment lines 38,39


if [ -d "/sys/bus/pci/devices/0000:01:00.0" ]
then
  working_switches=( '\_SB_.PCI0.PEG0.PEGP._DSM' '\_SB.PCI0.GFX0._DSM' '\_SB.PCI0.PEG0.PEGP._DSM' '\_SB.PCI0.PEG0.PEGP._OFF' )
  
  #echo '\_SB_.PCI0.PEG0.PEGP._DSM' > /proc/acpi/call 
  #echo '\_SB.PCI0.GFX0._DSM' > /proc/acpi/call 
  #echo '\_SB.PCI0.PEG0.PEGP._DSM' > /proc/acpi/call 
  echo '\_SB.PCI0.PEG0.PEGP._OFF' > /proc/acpi/call 
  echo -n 1 > '/sys/bus/pci/devices/0000:01:00.0/remove'

fi
```

</details>

I have tried running all of the commented out `echo` calls and then rebooting, but with no success.

## In intel mode:

<details>
<summary> `inxi -Fxxx --no-host` </summary>

```
System:    Kernel: 5.6.16-1-MANJARO x86_64 bits: 64 compiler: gcc v: 10.1.0 Desktop: KDE Plasma 5.18.5 tk: Qt 5.15.0 
           info: latte-dock wm: kwin_x11 dm: SDDM Distro: Manjaro Linux 
Machine:   Type: Laptop System: Razer product: Blade Pro 17 (Fall 2019) v: 3.04 serial: <root required> Chassis: type: 10 
           serial: <root required> 
           Mobo: Razer model: DA730 serial: <root required> UEFI: Razer v: 1.02 date: 08/22/2019 
Battery:   ID-1: BAT0 charge: 72.9 Wh condition: 72.9/70.6 Wh (103%) volts: 17.5/15.4 model: Razer Blade type: Unknown 
           serial: CNB1RC30-028798E00258-A03 status: Full 
CPU:       Topology: 6-Core model: Intel Core i7-9750H bits: 64 type: MT MCP arch: Kaby Lake rev: A L2 cache: 12.0 MiB 
           flags: avx avx2 lm nx pae sse sse2 sse3 sse4_1 sse4_2 ssse3 vmx bogomips: 62431 
           Speed: 1604 MHz min/max: 800/4500 MHz Core speeds (MHz): 1: 1604 2: 2217 3: 1506 4: 1728 5: 2144 6: 1931 7: 1790 
           8: 1248 9: 3270 10: 2313 11: 2967 12: 1373 
Graphics:  Device-1: Intel UHD Graphics 630 vendor: Razer USA driver: i915 v: kernel bus ID: 00:02.0 chip ID: 8086:3e9b 
           Device-2: NVIDIA TU104M [GeForce RTX 2080 Mobile] vendor: Razer USA driver: N/A bus ID: 01:00.0 chip ID: 10de:1e90 
           Display: x11 server: X.Org 1.20.8 driver: intel compositor: kwin_x11 resolution: 1920x1080~240Hz 
           OpenGL: renderer: Mesa Intel UHD Graphics 630 (CFL GT2) v: 4.6 Mesa 20.0.7 direct render: Yes 
Audio:     Device-1: Intel Cannon Lake PCH cAVS vendor: Razer USA driver: snd_hda_intel v: kernel bus ID: 00:1f.3 
           chip ID: 8086:a348 
           Device-2: NVIDIA TU104 HD Audio vendor: Razer USA driver: snd_hda_intel v: kernel bus ID: 01:00.1 
           chip ID: 10de:10f8 
           Sound Server: ALSA v: k5.6.16-1-MANJARO 
Network:   Device-1: Realtek RTL8125 2.5GbE vendor: Razer USA driver: r8169 v: kernel port: 3000 bus ID: 3d:00.0 
           chip ID: 10ec:8125 
           IF: enp61s0 state: down mac: 90:20:3a:1c:38:f1 
           Device-2: Intel Wi-Fi 6 AX200 driver: iwlwifi v: kernel port: 3000 bus ID: 3f:00.0 chip ID: 8086:2723 
           IF: wlp63s0 state: up mac: 90:78:41:30:f7:a5 
Drives:    Local Storage: total: 476.94 GiB used: 45.61 GiB (9.6%) 
           ID-1: /dev/nvme0n1 vendor: Samsung model: MZVLB512HAJQ-00000 size: 476.94 GiB speed: 31.6 Gb/s lanes: 4 
           serial: S3W8NA0M536264 rev: EXA7301Q temp: 44 C scheme: GPT 
           ID-2: /dev/nvme1n1 model: Sabrent size: 953.87 GiB speed: 31.6 Gb/s lanes: 4 serial: A0D60797047400005185 
           rev: ECFM12.3 temp: 29 C scheme: GPT
Partition: ID-1: / size: 920.78 GiB used: 45.61 GiB (5.0%) fs: ext4 dev: /dev/nvme1n1p2 
           ID-2: swap-1 size: 17.09 GiB used: 0 KiB (0.0%) fs: swap dev: /dev/nvme1n1p3 
Sensors:   System Temperatures: cpu: 67.0 C mobo: N/A 
           Fan Speeds (RPM): N/A 
Info:      Processes: 283 Uptime: 42m Memory: 15.54 GiB used: 2.35 GiB (15.2%) Init: systemd v: 245 Compilers: gcc: 10.1.0 
           alt: 8 Shell: bash v: 5.0.17 running in: yakuake inxi: 3.0.37
```
</details>

<details>
<summary>  `sudo /etc/switch/gpu_switch_check.sh` </summary>
                                       
```
Trying \_SB.PCI0.P0P1.VGA._OFF: /etc/switch/gpu_switch_check.sh: line 63: warning: command substitution: ignored null byte in input
failed
Trying \_SB.PCI0.P0P2.VGA._OFF: /etc/switch/gpu_switch_check.sh: line 63: warning: command substitution: ignored null byte in input
failed
Trying \_SB_.PCI0.OVGA.ATPX: /etc/switch/gpu_switch_check.sh: line 63: warning: command substitution: ignored null byte in input
failed
Trying \_SB_.PCI0.OVGA.XTPX: /etc/switch/gpu_switch_check.sh: line 63: warning: command substitution: ignored null byte in input
failed
Trying \_SB.PCI0.P0P3.PEGP._OFF: /etc/switch/gpu_switch_check.sh: line 63: warning: command substitution: ignored null byte in input
failed
Trying \_SB.PCI0.P0P2.PEGP._OFF: /etc/switch/gpu_switch_check.sh: line 63: warning: command substitution: ignored null byte in input
failed
Trying \_SB.PCI0.P0P1.PEGP._OFF: /etc/switch/gpu_switch_check.sh: line 63: warning: command substitution: ignored null byte in input
failed
Trying \_SB.PCI0.MXR0.MXM0._OFF: /etc/switch/gpu_switch_check.sh: line 63: warning: command substitution: ignored null byte in input
failed
Trying \_SB.PCI0.PEG1.GFX0._OFF: /etc/switch/gpu_switch_check.sh: line 63: warning: command substitution: ignored null byte in input
failed
Trying \_SB.PCI0.PEG0.GFX0.DOFF: /etc/switch/gpu_switch_check.sh: line 63: warning: command substitution: ignored null byte in input
failed
Trying \_SB.PCI0.PEG1.GFX0.DOFF: /etc/switch/gpu_switch_check.sh: line 63: warning: command substitution: ignored null byte in input
failed
Trying \_SB.PCI0.PEG0.PEGP._OFF: /etc/switch/gpu_switch_check.sh: line 63: warning: command substitution: ignored null byte in input
works!
Trying \_SB.PCI0.XVR0.Z01I.DGOF: /etc/switch/gpu_switch_check.sh: line 63: warning: command substitution: ignored null byte in input
failed
Trying \_SB.PCI0.PEGR.GFX0._OFF: /etc/switch/gpu_switch_check.sh: line 63: warning: command substitution: ignored null byte in input
failed
Trying \_SB.PCI0.PEG.VID._OFF: /etc/switch/gpu_switch_check.sh: line 63: warning: command substitution: ignored null byte in input
failed
Trying \_SB.PCI0.PEG0.VID._OFF: /etc/switch/gpu_switch_check.sh: line 63: warning: command substitution: ignored null byte in input
failed
Trying \_SB.PCI0.P0P2.DGPU._OFF: /etc/switch/gpu_switch_check.sh: line 63: warning: command substitution: ignored null byte in input
failed
Trying \_SB.PCI0.P0P4.DGPU.DOFF: /etc/switch/gpu_switch_check.sh: line 63: warning: command substitution: ignored null byte in input
failed
Trying \_SB.PCI0.IXVE.IGPU.DGOF: /etc/switch/gpu_switch_check.sh: line 63: warning: command substitution: ignored null byte in input
failed
Trying \_SB.PCI0.RP00.VGA._PS3: /etc/switch/gpu_switch_check.sh: line 63: warning: command substitution: ignored null byte in input
failed
Trying \_SB.PCI0.RP00.VGA.P3MO: /etc/switch/gpu_switch_check.sh: line 63: warning: command substitution: ignored null byte in input
failed
Trying \_SB.PCI0.GFX0.DSM._T_0: /etc/switch/gpu_switch_check.sh: line 63: warning: command substitution: ignored null byte in input
failed
Trying \_SB.PCI0.LPC.EC.PUBS._OFF: /etc/switch/gpu_switch_check.sh: line 63: warning: command substitution: ignored null byte in input
failed
Trying \_SB.PCI0.P0P2.NVID._OFF: /etc/switch/gpu_switch_check.sh: line 63: warning: command substitution: ignored null byte in input
failed
Trying \_SB.PCI0.P0P2.VGA.PX02: /etc/switch/gpu_switch_check.sh: line 63: warning: command substitution: ignored null byte in input
failed
Trying \_SB_.PCI0.PEGP.DGFX._OFF: /etc/switch/gpu_switch_check.sh: line 63: warning: command substitution: ignored null byte in input
failed
Trying \_SB_.PCI0.VGA.PX02: /etc/switch/gpu_switch_check.sh: line 63: warning: command substitution: ignored null byte in input
failed
Trying \_SB.PCI0.PEG0.PEGP.SGOF: /etc/switch/gpu_switch_check.sh: line 63: warning: command substitution: ignored null byte in input
failed
Trying \_SB.PCI0.AGP.VGA.PX02: /etc/switch/gpu_switch_check.sh: line 63: warning: command substitution: ignored null byte in input
failed
Trying \_SB.PCI0.PEG0.PEGP._DSM: /etc/switch/gpu_switch_check.sh: line 63: warning: command substitution: ignored null byte in input
works!
Trying \_SB.PCI0.RP01.PEGP._DSM: /etc/switch/gpu_switch_check.sh: line 63: warning: command substitution: ignored null byte in input
failed
Trying \_SB.PCI0.GFX0._DSM: /etc/switch/gpu_switch_check.sh: line 63: warning: command substitution: ignored null byte in input
works!
Trying \_SB_.PCI0.RP01.PXSX._DSM: /etc/switch/gpu_switch_check.sh: line 63: warning: command substitution: ignored null byte in input
failed
Trying \_SB.PCI0.PEG.VID._DSM: /etc/switch/gpu_switch_check.sh: line 63: warning: command substitution: ignored null byte in input
failed
Trying \_SB_.PCI0.PEG0.PEGP._DSM: /etc/switch/gpu_switch_check.sh: line 63: warning: command substitution: ignored null byte in input
works!
Trying \_SB.PCI0.AGP.VGA.PX02: /etc/switch/gpu_switch_check.sh: line 63: warning: command substitution: ignored null byte in input
failed
Trying \_SB.PCI0.PEG0.PEGP._DSM_OFF: /etc/switch/gpu_switch_check.sh: line 63: warning: command substitution: ignored null byte in input
failed
Trying \_SB.PCI0.RP01.PEGP._DSM_OFF: /etc/switch/gpu_switch_check.sh: line 63: warning: command substitution: ignored null byte in input
failed
Trying \_SB.PCI0.GFX0._DSM_OFF: /etc/switch/gpu_switch_check.sh: line 63: warning: command substitution: ignored null byte in input
failed
Trying \_SB_.PCI0.RP01.PXSX._DSM_OFF: /etc/switch/gpu_switch_check.sh: line 63: warning: command substitution: ignored null byte in input
failed
Trying \_SB.PCI0.PEG.VID._DSM_OFF: /etc/switch/gpu_switch_check.sh: line 63: warning: command substitution: ignored null byte in input
failed
Trying \_SB_.PCI0.PEG0.PEGP._DSM_OFF: /etc/switch/gpu_switch_check.sh: line 63: warning: command substitution: ignored null byte in input
failed
Trying \_SB.PCI0.AGP.VGA.PX02._DSM_OFF: /etc/switch/gpu_switch_check.sh: line 63: warning: command substitution: ignored null byte in input
failed
Trying \_SB.PCI0.PEG0.PEGP._DSM._OFF: /etc/switch/gpu_switch_check.sh: line 63: warning: command substitution: ignored null byte in input
failed
Trying \_SB.PCI0.RP01.PEGP._DSM._OFF: /etc/switch/gpu_switch_check.sh: line 63: warning: command substitution: ignored null byte in input
failed
Trying \_SB.PCI0.GFX0._DSM._OFF: /etc/switch/gpu_switch_check.sh: line 63: warning: command substitution: ignored null byte in input
failed
Trying \_SB_.PCI0.RP01.PXSX._DSM._OFF: /etc/switch/gpu_switch_check.sh: line 63: warning: command substitution: ignored null byte in input
failed
Trying \_SB.PCI0.PEG.VID._DSM._OFF: /etc/switch/gpu_switch_check.sh: line 63: warning: command substitution: ignored null byte in input
failed
Trying \_SB_.PCI0.PEG0.PEGP._DSM._OFF: /etc/switch/gpu_switch_check.sh: line 63: warning: command substitution: ignored null byte in input
failed
```
</details>

In summary, the script correctly disables nvidia drivers, and then brings them back properly when using prime mode. The only problem is that the GPU is not switched off, and still draws power.

**Comment by balbok0** [comment]
My bad. Package Manager update made me realize that I haven't fully uninstalled `optimus-manager` which was causing the issue.
